master/pkg/tasks/task.go

```diff
@@ -115,6 +117,12 @@ func (t TaskSpec) EnvVars() map[string]string {
 		"DET_ALLOCATION_ID": t.AllocationID,
 		"DET_SESSION_TOKEN": t.AllocationSessionToken,
 	}
+	if t.UserSessionToken != "" {
```

Comment:
question: when would we ever launch a container which didn't have a user and a token?  Shouldn't that always be the case?

Reply:
I learned it the hard way :) checkpoint gc is the answer.

Reply:
Can we give them the same user/token as an experiment?  That seems cleaner to me since checkpoint gc always belongs to one experiment which always belongs to one user.  If not, at least include a comment here explaining when these values are empty.

Reply:
After trying it for hours I could not reproduce the situation where the owner is not being set, even for a checkpoint gc. I am going to rely on it being there; if we run into problems downstream I will deal with them.